gwcs/wcs.py

```diff
@@ -794,6 +794,132 @@ def to_fits_sip(self, bounding_box=None, max_pix_error=0.25, degree=None,
         hdr['cd2_2'] = cdmat[1][1]
         return hdr
 
+    def to_fits_tab(self, bounding_box=None, bin_ext_name='WCS-TABLE',
+                    crd_col_name='coordinates', sampling=1):
+        """
+        Construct a FITS WCS ``-TAB``-based approximation to the WCS
+        in the form of a FITS header and a binary table extension. For the
+        description of the FITS WCS ``-TAB`` convention, see
+        "Representations of spectral coordinates in FITS" in
+        `Greisen, E. W. et al. A&A 446 (2) 747-771 (2006)
+        <https://doi.org/10.1051/0004-6361:20053818>`_ .
+
+        Parameters
+        ----------
+        bounding_box : tuple, optional
+            A pair of tuples, each consisting of two numbers
+            Represents the range of pixel values in both dimensions
+            ((xmin, xmax), (ymin, ymax))
+
+        bin_ext_name : str, optional
+            Extension name for the `~astropy.io.fits.BinTableHDU` extension.
+
+        crd_col_name : str, optional
+            Field name of the coordinate array in the structured array
+            stored in `~astropy.io.fits.BinTableHDU` data. This corresponds to
+            ``TTYPEi`` field in the FITS header of the binary table extension.
+
+        sampling : float, optional
+            The step, in input pixels, of the coordinate grid to be generated
+            for the coordinate array of the ``-TAB`` FITS WCS convention.
+
+        Returns
+        -------
+        hdr : astropy.io.fits.Header
+            Header with WCS-TAB information associated (to be used) with image
+            data.
+
+        bin_table : astropy.io.fits.BinTableHDU
+            Binary table extension containing the coordinate array.
+
+        Raises
+        ------
+        ValueError
+            When ``bounding_box`` is not defined either through the input
+            ``bounding_box`` parameter or this object's ``bounding_box``
+            property.
+
+        ValueError
+            When the number of ``sampling`` values is not 1 and also
+            does not match the number of image axes.
+
+        RuntimeError
+            If the number of image axes (`~gwcs.WCS.pixel_n_dim`) is larger
+            than the number of world axes (`~gwcs.WCS.world_n_dim`).
+
+        """
+        if bounding_box is None:
+            if self.bounding_box is None:
+                raise ValueError(
+                    "Need a valid bounding_box to compute the footprint."
+                )
+            bounding_box = self.bounding_box
+
```

Comment:
I suppose one should check to see if the value of bounding box parameter is as expected (this could be a useful utility function since I imagine it must be checked in multiple places)

Reply:
Well, I was just following `to_fits_sip()` 😄 This particular line should be "safe" since it is already a property of the `gwcs` object. But I agree about a function that validates the bounding box. @nden ?

Reply:
If `bounding_box` is not formatted correctly, the code would crash at https://github.com/spacetelescope/gwcs/pull/295/files/d1a97c3ef046ea087f556eab87963531ad5572df#diff-74d6657ed2fef3611c7388221e3776dbR875

Reply:
Don't follow what I do, but what I say :-) But you are right, we should be consistent in that. We could save that improvement for later.